assets/js/script_spotify.js

```diff
@@ -1,82 +1,115 @@
-  dataType: "json",
-  success: function (response) {
-    console.log(token);
-    console.log(response);
-  },
-  error: function (response) {
-    console.log(token);
-    console.log(response);
-  } 
-});
+function get_genresArray(token){
+  $.ajax({
+    type: "GET",
+    url: "https://api.spotify.com/v1/recommendations/available-genre-seeds",
+    headers: {
+      'Authorization': "Bearer " + token
+    },
+    contentType: "application/json",
+    async: false,
+    dataType: "json",
+    success: function (response) {
+      console.log("genreArray token", token);
+      console.log("genreArray response", response);
+      get_Tracks(token, response);
+    },
+    error: function (response) {
+      console.log("error", token);
+      console.log("error", response);
+    } 
+  });
+}
 
 // Gets List of tracks based on selected genre
-$.ajax({
-  type: "GET",
-  url: "https://api.spotify.com/v1/recommendations",
-  headers: {
-    'Authorization': "Bearer " + token
-  },
-  data: {
-    seed_genres: "alternative" // dynamic variable set by user input
-  },
-  contentType: "application/json",
-  async: false,
-  dataType: "json",
-  success: function (response) {
-    console.log(response);
-    tracks = response.tracks;
-    console.log(tracks);
-  }
-});
+function get_Tracks(token, genresArr){
+  var chosenGenre = get_Genre(genresArr);
+  console.log("chosenGenre", chosenGenre);
+  $.ajax({
+    type: "GET",
+    url: "https://api.spotify.com/v1/recommendations",
+    headers: {
+      'Authorization': "Bearer " + token
+    },
+    data: {
+      seed_genres: chosenGenre // dynamic variable set by user input
+    },
+    contentType: "application/json",
+    async: false,
+    dataType: "json",
+    success: function (response) {
+      console.log("GET TRACKS", response);
+      tracks = response.tracks;
+      console.log("TRACKS", tracks);
+      get_TrackId(tracks);
+    },
+    error: function(response){
+      console.log("error", response);
+    }
+  });
+}
+
+//choose a genre from user's input (randomly generates for now)
+//genresArr param might not be necessary, was using it to randomly generate genres -- Hailey F.
+function get_Genre(genresArr){
+  var userInput = $("#musicSearch").children("input").val();
+  console.log("userInput", userInput);
+  //TODO: get the user's input to choose a genre
+    // I believe we were talking about using a dropdown menu on the HTML side? -- Hailey F.
+  
+
+  //substitute: generates random genre
+  // var randomNum = Math.floor(Math.random() * 126);
+  // console.log("randomNum", randomNum);
+  // console.log("genreChosen", genresArr.genres[randomNum]);
+  return userInput;
+}
+
+//gets trackID from chosen tracks array
+function get_TrackId(tracks) {
+  var randomNum = Math.floor(Math.random() * 20);
+  console.log(randomNum);
+  track_id = tracks[randomNum].id;
+  appendTrack(track_id);
+};
+
+// Creates iFrame element with dynamically selected track id
+function appendTrack(track_id){
```

Comment:
Needs an empty methond on the music-player element so that the element does not contain two embeds at the same time.

Reply:
Added empty method on line 113, let me know if this is an acceptable solution